mijia-homie/src/main.rs

```diff
@@ -310,14 +317,22 @@ async fn bluetooth_connection_loop(
 
 #[derive(Debug)]
 struct SensorState {
-    sensors: HashMap<DeviceId, Sensor>,
+    sensors: HashMap<MacAddress, Sensor>,
     homie: HomieDevice,
 }
 
+/// Get the sensor entry for the given id, if any.
+fn get_mut_sensor_by_id<'a>(
+    sensors: &'a mut HashMap<MacAddress, Sensor>,
+    id: &DeviceId,
+) -> Option<&'a mut Sensor> {
+    sensors.values_mut().find(|sensor| sensor.ids.contains(id))
```

Comment:
can we do `sensors.get_mut(id.mac_address())` or something inline and kill off this function?

Reply:
I'm still not really convinced that parsing BlueZ D-Bus paths is the way to go, it feels quite brittle.